src/prefect/cache_policies.py

```diff
@@ -219,7 +219,7 @@ def compute_key(
         except TypeError:
             lines = inspect.getsource(task_ctx.task.fn.__class__)
         except OSError as exc:
-            if "could not get source code" in str(exc):
+            if str(exc) in {"could not get source code", "source code not available"}:
```

Comment:
the reversal of containment here won't work, as `str(exc)` is a longer message than these two snippets.  I suggest simplifying this to:
```suggestion
            if "source code" in str(exc):
```

Reply:
In this case the message does seem to match the strings directly, at least from my local testing & the test case below:

```python
In [3]: import inspect

In [4]: class Test:
   ...:     pass
   ...:

In [5]: try:
   ...:     inspect.getsource(Test)
   ...: except OSError as e:
   ...:     print(str(e))
   ...:
source code not available
```

But matching just the `source code` snippet is much nicer/simpler - I went with the set of full messages because I wasn't sure if there were any other `OSError` instances raisable by `inspect.getsource` which would get matched and handled incorrectly, but that doesn't seem to be the case.

Reply:
Ah yea you're right! Sorry about that - but yea, the simpler check should be more robust / general so thanks for updating still 🙏